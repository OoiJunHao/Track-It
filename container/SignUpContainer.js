import React from "react";
import TextBox from "../component/TextBox";
import {
  StyleSheet,
  View,
  Image,
  TextInput,
  Text,
  KeyboardAvoidingView,
  ScrollView,
  TouchableWithoutFeedback,
  Keyboard,
  SafeAreaView,
  BackHandler,
  TouchableOpacity,
} from "react-native";
import BackArrow from "../component/BackArrow";
import Button from "../component/Button";
import firebaseDb from "../firebaseDb";
import PasswordTextBox from "../component/PasswordTextBox";
import Icons from "react-native-vector-icons/MaterialIcons";

class SignUpContainer extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    password2: "",
  };

  handleUpdateName = (name) => this.setState({ name });
  handleUpdateEmail = (email) => this.setState({ email });
  handleUpdatePassword = (password) => this.setState({ password });
  handleUpdatePassword2 = (password2) => this.setState({ password2 });
  handleCreateUser = () => {
    firebaseDb
      .firestore()
      .collection("users")
      .add({
        name: this.state.name,
        email: this.state.email,
        password: this.state.password,
      })
      .then(() => {
        this.setState({
          name: "",
          email: "",
          password: "",
          password2: "",
        });
        this.props.navigation.navigate("Home");
      })
      .catch((err) => console.error(err));
  };

  handleRegister = () => {
    Keyboard.dismiss();
    if (
      this.state.name.length &&
      this.state.email.length &&
      this.state.password.length &&
      this.state.password2.length &&
      this.state.password == this.state.password2 &&
      this.state.email.includes("@")
    ) {
      this.handleCreateUser();
    } else if (
      !this.state.name.length ||
      !this.state.email.length ||
      !this.state.password.length ||
      !this.state.password2.length
    ) {
      alert("Please fill in all fields");
      console.log("Empty field");
    } else if (this.state.password != this.state.password2) {
      alert("Passwords do not match");
      console.log("Non matching passwords");
    } else if (!this.state.email.includes("@")) {
      alert("Invalid Email");
      console.log("Invalid Email");
    }
  };

  handleBackButtonClick = () => {
    Keyboard.dismiss();
    this.props.navigation.goBack();
  };

  render() {
    return (
      <SafeAreaView style={{ flex: 1 }}>
        <TouchableWithoutFeedback
          onPress={() => {
            Keyboard.dismiss();
            console.log("dismissed keyboard");
          }}
        >
          <KeyboardAvoidingView behavior="padding" style={styles.container}>
            <ScrollView
              contentContainerStyle={{
                flex: 1,
              }}
            >
              <View>
                <TouchableOpacity onPress={this.handleBackButtonClick}>
                  <Icons
                    name={"arrow-back"}
                    size={30}
                    color="black"
                    style={{ marginLeft: "3%", marginTop: "16%" }}
                  />
                </TouchableOpacity>
              </View>
              <View style={styles.logoContainer}>
                <Text
                  style={{ fontSize: 35, padding: "5%", fontWeight: "bold" }}
                >
                  Let's get started!
                </Text>
                <Text>Create an account to use our features</Text>
                <View style={styles.inputContainer}>
                  <TextBox
                    icon="person"
                    label="Username"
                    style={(styles.input, { marginBottom: 25, padding: 10 })}
                    onChange={this.handleUpdateName}
                    value={this.state.name}
                    returnKeyType="next"
                    onSubmitEditing={() => this.emailInput.focus()}
                  />
                  <TextBox
                    icon="mail"
                    label="Email"
                    style={(styles.input, { marginBottom: 25, padding: 10 })}
                    onChange={this.handleUpdateEmail}
                    value={this.state.email}
                    returnKeyType="next"
                    onSubmitEditing={() => this.firstPasswordInput.focus()}
                    keyboardType="email-address"
                  />
                  <PasswordTextBox
                    icon="lock"
                    label="Password"
                    onChange={this.handleUpdatePassword}
                    value={this.state.password}
                    returnKeyType="next"
                    style={[styles.input, { marginBottom: 25, padding: 10 }]}
                    ref={(input) => (this.firstPasswordInput = input)}
                    onSubmitEditing={() => this.secondPasswordInput.focus()}
                  />
                  <PasswordTextBox
                    icon="lock"
                    label="Re-enter Password"
                    onChange={this.handleUpdatePassword2}
                    value={this.state.password2}
                    returnKeyType="go"
                    style={[styles.input, { marginBottom: 25, padding: 10 }]}
                    ref={(input) => (this.secondPasswordInput = input)}
                  />
                  <Button
                    onPress={this.handleRegister}
                    style={styles.buttonContainer}
                  >
                    <Text>Register</Text>
                  </Button>
                </View>
              </View>
            </ScrollView>
          </KeyboardAvoidingView>
        </TouchableWithoutFeedback>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    backgroundColor: "white",
    alignItems: "center",
    justifyContent: "center",
  },
  logoContainer: {
    alignItems: "center",
    flexGrow: 1,
    justifyContent: "center",
  },
  logo: {
    width: 270,
    height: 110,
    resizeMode: "contain",
  },
  inputContainer: {
    padding: 20,
    width: 360,
    marginTop: 16,
  },
  input: {
    height: 40,
    width: 270,
    backgroundColor: "#d7dedc",
    marginBottom: 8,
    color: "black",
    paddingHorizontal: 10,
    borderColor: "gray",
    borderWidth: 1,
  },
  buttonContainer: {
    backgroundColor: "#2980b9",
    paddingVertical: 15,
    marginBottom: 10,
    marginTop: 16,
  },
  text: {
    fontSize: 20,
    color: "red",
    marginTop: 48,
  },
});

export default SignUpContainer;
