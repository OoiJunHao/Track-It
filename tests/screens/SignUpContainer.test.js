import React from 'react';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import renderer from 'react-test-renderer'
import SignUpContainer from '../../container/SignUpContainer';
import MockFirebase from 'firebase-mock';

afterEach(cleanup);

describe('SignUpContainer', () => {
    it('matched snapshot', () => {
        const rendered = renderer.create(<SignUpContainer />).toJSON();
        expect(rendered).toMatchSnapshot();
    })

    it('should receive the password value', () => {
        let signUpComponent = renderer.create(<SignUpContainer />).getInstance()
        signUpComponent.handleUpdatePassword('123456')
        const user = signUpComponent.state.user
        expect(user.password).toEqual('123456')
    })

    it('should receive the email value', () => {
        let signUpComponent = renderer.create(<SignUpContainer />).getInstance()
        signUpComponent.handleUpdateEmail('123456')
        const user = signUpComponent.state.user
        expect(user.email).toEqual('123456')
    })
    it('should receive the password2 value', () => {
        let signUpComponent = renderer.create(<SignUpContainer />).getInstance()
        signUpComponent.handleUpdatePassword2('123456')
        const user = signUpComponent.state.user
        expect(user.password2).toEqual('123456')
    })
    it('should receive the name value', () => {
        let signUpComponent = renderer.create(<SignUpContainer />).getInstance()
        signUpComponent.handleUpdateName('123456')
        const user = signUpComponent.state.user
        expect(user.name).toEqual('123456')
    })

})
