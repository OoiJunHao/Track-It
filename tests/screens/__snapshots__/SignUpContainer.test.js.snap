// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpContainer matched snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "resizeMode": "contain",
              "top": -170,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(21,22,48,0.1)",
              "borderRadius": 16,
              "height": 32,
              "justifyContent": "center",
              "left": 17,
              "opacity": 1,
              "position": "absolute",
              "top": 10,
              "width": 32,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "position": "absolute",
              "top": 50,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 35,
                "fontWeight": "bold",
                "padding": "5%",
              }
            }
          >
            Let's get started!
          </Text>
          <Text>
            Create an account to use our features
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 48,
              "marginHorizontal": 30,
              "marginTop": "50%",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
              }
            }
            testID="wrapper"
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
              testID="text"
            >
              Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              testID="input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
              }
            }
            testID="wrapper"
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
              testID="text"
            >
              Email
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              testID="input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "marginBottom": 30,
                "marginTop": 32,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#8A8F9E",
                  "fontSize": 10,
                  "textTransform": "uppercase",
                }
              }
            >
              Re-enter Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#8A8F9E",
                  "borderBottomWidth": 0.5,
                  "color": "#161f3d",
                  "fontSize": 15,
                  "height": 40,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#70bee2",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginHorizontal": 30,
                "opacity": 1,
                "paddingVertical": 15,
              }
            }
            testID="submit"
          >
            <Text
              style={
                Object {
                  "color": "rgb(108,108,108)",
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
              testID="text"
            >
              <Text>
                Sign Up
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
